vault/plugin_catalog.go

```diff
@@ -140,6 +143,37 @@ func (p *pluginClient) Conn() grpc.ClientConnInterface {
 	return p.clientConn
 }
 
+func (p *pluginClient) Reload() error {
+	p.logger.Debug("reload external plugin process")
+	return p.reloadFunc()
+}
+
+// reloadExternalPlugin
+// This should be called with the write lock held.
+func (c *PluginCatalog) reloadExternalPlugin(name, id string) error {
```

Comment:
The name reload confuses me a bit, because I only see this function killing the process, but not starting it again. I'm sure it's just me missing context, but I'd love to understand this bit of the lifecycle a bit better.

Reply:
Yeah, good point. You are right that it doesn't actually do any reloading itself. It is used in the reload path though.

This is a product of the current reload mechanism, which I am a bit unhappy with. But I haven't come up with anything better. I will think about how this name could be improved.